datastore/modelassertmanager.go

```diff
@@ -0,0 +1,154 @@
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package datastore
+
+import (
+	"errors"
+	"log"
+	"time"
+)
+
+const createModelAssertTableSQL = `
+	CREATE TABLE IF NOT EXISTS modelassertion (
+		id               serial primary key not null,
+		model_id         int references model not null,
+		keypair_id       int references keypair not null,
+		series           int not null,
+		architecture     varchar(20) not null,
+		revision         int not null default 0,
+		gadget           varchar(60) not null,
+		kernel           varchar(60) not null,
+		store            varchar(60),
+		created          timestamp default current_timestamp,
+		modified         timestamp default current_timestamp
+	)
+`
+const createModelAssertSQL = "INSERT INTO modelassertion (model_id,keypair_id,series,architecture,revision,gadget,kernel,store) VALUES ($1,$2,$3,$4,$5,$6,$7,$8) RETURNING id"
+
+const updateModelAssertSQL = `
+UPDATE modelassertion
+SET model_id=$2, keypair_id=$3, series=$4, architecture=$5, revision=$6, gadget=$7, kernel=$8, store=$9, modified=$10
+WHERE id=$1`
+
+const getModelAssertSQL = `
+SELECT id,model_id,keypair_id,series,architecture,revision,gadget,kernel,store,created,modified
+FROM modelassertion
+WHERE model_id=$1
+`
+
+// ModelAssertion holds the model assertion details in the local database
+type ModelAssertion struct {
+	ID           int       `json:"id"`
+	ModelID      int       `json:"model_id"`
+	KeypairID    int       `json:"keypair_id"`
+	Series       int       `json:"series"`
+	Architecture string    `json:"architecture"`
+	Revision     int       `json:"revision"`
+	Gadget       string    `json:"gadget"`
+	Kernel       string    `json:"kernel"`
+	Store        string    `json:"store"`
+	Created      time.Time `json:"created"`
+	Modified     time.Time `json:"modified"`
+}
+
+// CreateModelAssertTable creates the database table for a model assertion
+func (db *DB) CreateModelAssertTable() error {
+	_, err := db.Exec(createModelAssertTableSQL)
+	return err
+}
+
+// CreateModelAssert adds a model assertion record to allow generation of a signed assertion
+func (db *DB) CreateModelAssert(m ModelAssertion) (int, error) {
+	var createdID int
+	err := db.QueryRow(createModelAssertSQL, m.ModelID, m.KeypairID, m.Series, m.Architecture, m.Revision, m.Gadget, m.Kernel, m.Store).Scan(&createdID)
+	if err != nil {
+		log.Printf("Error creating the model assertion: %v\n", err)
+	}
+	return createdID, err
+}
+
+// UpdateModelAssert updates the model assertion details
+func (db *DB) UpdateModelAssert(m ModelAssertion) error {
+
```

Comment:
Cosmetic: not blank first line, or at least aligned with other methods if you want to maintain it